libraries/tock-register-interface/src/macros.rs

```diff
@@ -1,12 +1,20 @@
 //! Macros for cleanly defining peripheral registers.
 
+/// Helper macro for computing bitmask of variable number of bits
+#[macro_export]
+macro_rules! bitmask {
+    ($numbits:expr) => {
+        (1 << ($numbits - 1)) + ((1 << ($numbits - 1)) - 1)
```

Comment:
Would this be clearer as `!(!0 << $numbits)`? Or rather `!(!1 << ($numbits - 1))` to support the case where `$numbits` is equal to the size of the integer?

Reply:
`!(!0 << $numbits)` is not equivalent to the existing code when `$numbits = 32` and compiled in release mode (see this rust playground example: https://play.rust-lang.org/?version=nightly&mode=release&edition=2018&gist=bd7681e2e730ce2830fe9b14523cffe8).

`!(!1 << ($numbits - 1))` is equivalent to the existing code for all values of numbits in the range (1..64) when compiled in the rust playground in release mode, but it makes me slightly uneasy that the current code panics for `$numbits = 32` when compiled with debug_assertions=true and this new example does not -- does that mean they are not guaranteed to be equivalent to the existing code in the case where `$numbits = 32`?

It seems like `!(!1 << ($numbits - 1))` might be more correct than the current code, but I don't totally understand the current code and am wary of adding a commit that appears to introduce a possible logic change when we may be relying on the existing behavior somehow

Reply:
Yes, the `!(!0 << $numbits)` formula only works for `$numbits` in `0..32`, whereas the other two work for `$numbits` in `1..=32` (assuming a `u32`). This is because shifting by 32 bits is undefined behavior (contrary to what one may expect of it "clearing" all bits, some/most CPUs will mask the shift and shift zero bits instead).

I don't think we consider the case where `$numbits` is zero (which would be quite uninteresting), so the current formula is fine, and so would be `!(!1 << ($numbits - 1))`. I can propose a separate PR for that.